<launch>
  <arg name="param_path" default="$(find-pkg-share autoware_rear_obstacle_checker)/config/rear_obstacle_checker.param.yaml"/>

  <arg name="input_objects" default="/perception/object_recognition/objects"/>
  <arg name="input_odometry" default="/localization/kinematic_state"/>
  <arg name="input_acceleration" default="/localization/acceleration"/>
  <arg name="input_pointcloud" default="/perception/obstacle_segmentation/pointcloud"/>
  <arg name="input_route_topic_name" default="/planning/mission_planning/route"/>
  <arg name="output_velocity_limit" default="/planning/scenario_planning/max_velocity_candidates"/>
  <arg name="output_velocity_limit_clear_command" default="/planning/scenario_planning/clear_velocity_limit"/>
  <arg name="input_vector_map_topic_name" default="/map/vector_map"/>

  <node pkg="autoware_rear_obstacle_checker" exec="autoware_rear_obstacle_checker_node" name="rear_obstacle_checker" output="screen">
    <param from="$(var param_path)"/>
    <remap from="~/input/pointcloud" to="$(var input_pointcloud)"/>
    <remap from="~/input/objects" to="$(var input_objects)"/>
    <remap from="~/input/odometry" to="$(var input_odometry)"/>
    <remap from="~/input/acceleration" to="$(var input_acceleration)"/>
    <remap from="~/input/lanelet_map_bin" to="$(var input_vector_map_topic_name)"/>
    <remap from="~/input/route" to="$(var input_route_topic_name)"/>
    <remap from="~/input/path_with_lane_id" to="/planning/scenario_planning/lane_driving/behavior_planning/path_with_lane_id"/>
    <remap from="~/input/trajectory" to="/planning/scenario_planning/trajectory"/>
  </node>
</launch>
